src/diffusers/models/unet_2d_blocks.py

```diff
@@ -407,15 +416,29 @@ def __init__(
 
         for _ in range(num_layers):
             if self.add_attention:
-                attentions.append(
-                    AttentionBlock(
+                if attention_block_type == "AttentionBlock":
+                    attention = AttentionBlock(
                         in_channels,
                         num_head_channels=attn_num_head_channels,
                         rescale_output_factor=output_scale_factor,
                         eps=resnet_eps,
                         norm_num_groups=resnet_groups,
                     )
-                )
+                elif attention_block_type == "Attention":
+                    attention = Attention(
+                        in_channels,
+                        heads=in_channels // attn_num_head_channels if attn_num_head_channels is not None else 1,
+                        dim_head=attn_num_head_channels if attn_num_head_channels is not None else in_channels,
+                        bias=True,
+                        upcast_softmax=True,
+                        norm_num_groups=resnet_groups,
+                        processor=SpatialAttnProcessor(),
```

Comment:
Hmm is there really a difference between Spatial Attention and "normal" attention - isn't it just some reshaping before and after that's different? 

Could we maybe try to use the normal attention processor and do the reshaping outside of the attention (e.g. in the VAE directly) - this would also allow us to use all the other attention processors out of the box (e.g. LoRA, xfomers, torch 2.0) 

Wdyt?

Reply:
wip commit up https://github.com/huggingface/diffusers/pull/2697/commits/5aa3446aaa029a9548fd195150d4c002974130f4